rosidl_runtime_cpp/include/rosidl_runtime_cpp/traits.hpp

```diff
@@ -15,11 +15,137 @@
 #ifndef ROSIDL_RUNTIME_CPP__TRAITS_HPP_
 #define ROSIDL_RUNTIME_CPP__TRAITS_HPP_
 
+#include <codecvt>
+#include <string>
 #include <type_traits>
 
 namespace rosidl_generator_traits
 {
 
+void value_to_yaml(bool value, std::ostream & out)
+{
+  out << (value ? "true" : "false");
+}
+
+void character_value_to_yaml(unsigned char value, std::ostream & out)
+{
+  auto flags = out.flags();
+  out << "0x" << std::hex << std::setw(2) << std::setfill('0') << +value;
```

Comment:
why the `+` prefixing `value`?

Reply:
Because the value needs to be streamed as a numeric value.

Reply:
I didn't know that trick, it was hard to find a place explaining it.

Reply:
See e46ba0ca046d13cc1b6a2b05f3d166921f9671c8 which explicitly casts to `uint16_t` to make it easier to read.